Tests/CarthageKitTests/DependencySpec.swift

```diff
@@ -84,6 +84,17 @@ class DependencySpec: QuickSpec {
 
 					expect(dependency.name) == "whatisthisurleven"
 				}
+
+                it("should be the directory name if the given URL is relative local path") {
+                    let fileManager = FileManager.default
+                    let previousDirectory = fileManager.currentDirectoryPath
+                    defer { fileManager.changeCurrentDirectoryPath(previousDirectory) }
+                    let temporaryDirectoryURL = URL(fileURLWithPath: NSTemporaryDirectory())
```

Comment:
I'm a tiny bit concerned about relying on the location of `NSTemporaryDirectory()`. In specific, this code is assuming that there will always be a directory below the temporary directory, right? Is that a safe assumption?

Reply:
I'm afraid I don't understand your concern precisely.

IMO it's enough because [the document of `NSTemporaryDirectory`](https://developer.apple.com/documentation/foundation/1409211-nstemporarydirectory) says
> Returns the path of the temporary directory for the current user.

In Darwin, `NSTemporaryDirectory` just returns the value defined in `confstr(_CS_DARWIN_USER_TEMP_DIR)` and fallbacks to `$TMPDIR` environment variable or `/tmp` (see [the implementation in Swift corelibs](https://github.com/apple/swift-corelibs-foundation/blob/3b471931ee535e8138bdada2f05ebb42095a7ed2/Foundation/NSPathUtilities.swift#L28-L57)).

So technically speaking, you can cheat `NSTemporaryDirectory` by setting `$TMPDIR` to the wrong path like a path which doesn't exist or a path to a regular file.
However that kind of settings violates to [the POSIX specification](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html) so I guess we don't have to consider the situation.

I don't know much about Windows but I suppose it may be the similar situation to Darwin.

Reply:
It probably isn't something we need to worry about but the concern is that in a different environment the temporary directory might be at the root of the directory structure. If that were the case I think the test would break?

But really it is a tiny concern and not likely to happen. Thanks for taking the time to respond.

Reply:
> the temporary directory might be at the root of the directory structure.

Honestly I don't know whether if the environment like that exists or not.
But if it exists and it complies to POSIX, which means the root directory is the temporary directory where any users can move it, so I guess it won't break the test.
But if it exists and it is non-POSIX OS, I don't know what happens (again, I don't have a good example for this situation).

BTW I found Carthage already uses `NSTemporaryDirectory` in another place.

https://github.com/Carthage/Carthage/blob/7cb0e6ab571f0ce32e9f94bf9a24cb80371be226/Source/carthage/Build.swift#L184

So I think we don't have to care about the use of it.